### PR TITLE
Idl reduction xcdr test

### PR DESCRIPTION
**Pull request by ClaytonCalabrese**
XCDR test was still timing out on remedy fedora so it requires addition splitting of its idl.

In the process of doing this I found a bug in the TS generation and fixed it.

### CODE REVIEW

File: tests/DCPS/Compiler/xcdr/appendable_mixed.idl [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#include <tao/LongSeq.pidl>
+#include <tao/ShortSeq.pidl>
+#include <tao/OctetSeq.pidl>
+
+#include <xcdrbasetypes.idl>
```
Comment (iguessthislldo) [inline tests/DCPS/Compiler/xcdr/appendable_mixed.idl:5]:
```suggestion
#include "xcdrbasetypes.idl"
```

Also should go above TAO includes.

File: tests/DCPS/Compiler/xcdr/mutable_types2.idl [hunk 2]
```diff
@@ -0,0 +1,77 @@
+#include <tao/LongSeq.pidl>
+#include <tao/ShortSeq.pidl>
+#include <tao/OctetSeq.pidl>
+#include <tao/StringSeq.pidl>
+
+#include <mutable_types.idl>
+#include <xcdrbasetypes.idl>
```
Comment (iguessthislldo) [inline tests/DCPS/Compiler/xcdr/mutable_types2.idl:7]:
```suggestion
#include "mutable_types.idl"
#include "xcdrbasetypes.idl"

#include <tao/LongSeq.pidl>
#include <tao/ShortSeq.pidl>
#include <tao/OctetSeq.pidl>
#include <tao/StringSeq.pidl>
```

File: tests/DCPS/Compiler/xcdr/keyonly.idl [hunk 3]
```diff
@@ -3,145 +3,7 @@
...
-};
-
-@mutable
-struct NestingMutableStruct {
-  string string_field;
-  AppendableStruct appendable_nested;
-  sequence<octet> sequence_field;
-  FinalStruct final_nested;
-};
+#include <xcdrbasetypes.idl>
```
Comment (iguessthislldo) [inline tests/DCPS/Compiler/xcdr/keyonly.idl:6]:
```suggestion
#include "xcdrbasetypes.idl"
```

Also should go above TAO includes.

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

**Review by iguessthislldo — APPROVED** [review]

**Comment by ClaytonCalabrese** [comment]
Looks like some stuff is broken due to the fix. I'll look into them.

**Pull request merged**